src/index.ts

```diff
@@ -0,0 +1,168 @@
+import * as _ from 'underscore';
+import { AWSProvider, ServerlessInstance, ServerlessConfig } from './types';
+
+interface PluginConfig {
+  stages?: string[];
+  roleArn?: string;
+  filterPattern: string;
+  normalizedFilterID: boolean;
+  createLambdaPermission: boolean;
+  destinationARN: string;
+}
+
+const configDefaults = {
+  filterPattern: '',
+  normalizedFilterID: true,
+  createLambdaPermission: true,
+};
+
+interface CFObject<TProps> {
+  Type: string;
+  DependsOn?: string[];
+  Properties: TProps;
+}
+
+interface LambdaPermissionProps {
+  Action: string;
+  Principal: string;
+  FunctionName: string;
+}
+
+interface SubscriptionFilterProps {
+  DestinationArn: string;
+  FilterPattern: string;
+  LogGroupName: string;
+  RoleArn?: string;
+}
+
```

Comment:
I would suggest moving interfaces to the `types.ts`.
Those are abstractions and can be used in different places.

Reply:
addressed